### PR TITLE
audacious 3.8.2 (new formula)

### PR DESCRIPTION
**Pull request by ibrokemypie**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/audacious.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    resource("audacious-plugins").stage do
+      ENV.prepend "PKG_CONFIG_PATH", "#{lib}/pkgconfig:"
+      system "./configure", "--prefix=#{prefix}"
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    system "false"
```
Comment (bfontaine) [inline Formula/audacious.rb]: Please write a test.

File: Formula/audacious.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    sha256 "d7cefca7a0e32bf4e58bb6e84df157268b5e9a6771a0e8c2da98b03f92a5fdd4"
+  end
+
+  def install
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+
+    resource("audacious-plugins").stage do
+      ENV.prepend "PKG_CONFIG_PATH", "#{lib}/pkgconfig:"
```
Comment (bfontaine) [inline Formula/audacious.rb]: Use `ENV.prepend_path "PKG_CONFIG_PATH", "#{lib}/pkgconfig"` instead.

**Review by bfontaine — CHANGES_REQUESTED** [review]

**Comment by ibrokemypie** [comment]
any ideas why head wont build? seems like the same environment variables arent being set. I opened a thread with more detail over here https://discourse.brew.sh/t/formula-fails-with-head-but-succeeds-with-the-same-content-without-head/718/4

File: Formula/audacious.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  def caveats; <<-EOS.undent
+    audtool does not work due to a broken dbus implementation on macOS, so is not built
+    coreaudio output has been disabled as it does not work (Fails to set audio unit input property.)
+    GTK+ gui is not built by default as the QT gui has better integration with macOS, and when built, the gtk gui takes precedence
+    EOS
+  end
+
+  test do
+    system bin/"audacious", "-H", "-V", "-q", "-E", test_fixtures("test.wav"), "-p"
+    system "true"
```
Comment (bfontaine) [inline Formula/audacious.rb]: You can remove this line now.

File: Formula/audacious.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
+class Audacious < Formula
+  desc "Free and advanced audio player based on GTK+"
+  homepage "http://audacious-media-player.org"
+  url "http://distfiles.audacious-media-player.org/audacious-3.8.2.tar.bz2"
+  sha256 "bdf1471cce9becf9599c742c03bdf67a2b26d9101f7d865f900a74d57addbe93"
+  head "https://github.com/audacious-media-player/audacious.git"
+
+  option "with-app", "Build .app"
```
Comment (bfontaine) [inline Formula/audacious.rb]: We don’t build `.app` files in Homebrew. Use Homebrew Cask for that.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/audacious.rb (hunk 4 above)
Comment (ibrokemypie) [inline Formula/audacious.rb]: cask does not support building from source, which is the only options as there are no .apps provided from upstream

Comment (bfontaine) [inline Formula/audacious.rb]: True. Still, please remove this option. Quoting from our ["Acceptable Formulae" guidelines](http://docs.brew.sh/Acceptable-Formulae.html):

> Don’t make your formula build an .app (native macOS Application); we don’t want those things in Homebrew (but Homebrew Cask does). Make it build a command-line tool or a library.

**Comment by fxcoudert** [comment]
Description says *“based on GTK+”* but it's actually built with Qt?

**Comment by ibrokemypie** [comment]
By default and originally it is a GTK app, however the QT build works better on macOS, which is why it builds with QT in this formula

**Review by bfontaine — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @ibrokemypie for the pull request, this has now been committed!

**Comment by ilovezfs** [comment]
The test doesn't work, and hangs until the test block itself times out after 5 minutes, and even then a process is left running permanently. (I just manually killed more than 12 of them on the CI nodes.)

It seems the test never worked including in the original CI (https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/1478/version=sierra/testReport/junit/brew-test-bot/sierra/test_audacious/).

As a stop-gap I'm at least temporarily switching this over to a simple test in https://github.com/Homebrew/homebrew-core/pull/13852 rather than removing the formula, but it would be good to get a non-trivial test working at some point.

**Pull request closed without merging**